src/feature/hs/hs_service.c

```diff
@@ -3404,6 +3404,29 @@ service_encode_descriptor(const hs_service_t *service,
   return ret;
 }
 
+/* An introduction circuit has timed out while building or waiting for an
+ * established introduction cell. */
+static void
+service_intro_circuit_timed_out(const origin_circuit_t *circ)
+{
+  hs_service_t *service = NULL;
+  hs_service_intro_point_t *ip = NULL;
+
+  tor_assert(circ);
+  tor_assert(TO_CIRCUIT(circ)->purpose == CIRCUIT_PURPOSE_S_ESTABLISH_INTRO ||
+             TO_CIRCUIT(circ)->purpose == CIRCUIT_PURPOSE_S_INTRO);
```

Comment:
Hmm, `CIRCUIT_PURPOSE_S_INTRO` should not be possible according to the caller. Is there a reason for having it here?

Reply:
Removed in fixup: ce79052eab

Reply:
Hm, I don't see it in the fixup.

Reply:
Oh my mistake. Fixup of the fixup: a05f3dd02d